src/main/webapp/script.js

```diff
@@ -220,51 +220,117 @@ function getGraphDisplay(graphNodes, graphEdges) {
     node.tip.show();
   });
 
-  const searchElement = document.getElementById('search');
-  document.getElementById('search-button').onclick = function() {
-    if (searchNode(cy, searchElement.value) || searchElement.value == "") {
-      document.getElementById('search-error').innerText = "";
+  document.getElementById('reset').onclick = function(){ resetElements(cy) };
+
+  document.getElementById('search-button').onclick = function() { search(cy, "node", searchNode) };
+
+  document.getElementById('search-token-button').onclick = function() { search(cy, "token", searchToken) };
+}
+
+/**
+ * Searches based on type (node or token)
+ */
+function search(cy, type, searchFunction) {
+  resetElements(cy);
+  let errorText = "";
+  let query = document.getElementById(type + '-search').value;
+  let result;
+  if (query != "") {
+    result = searchFunction(cy, query);
+    if (result) {
+      errorText = "";
     } else {
-      document.getElementById('search-error').innerText = "Node does not exist.";
+      errorText = type + " does not exist.";
     }
-  };
-  
+  }
+  document.getElementById(type + '-error').innerText = errorText;
+  return result;
 }
 
 /**
- * Zooms in on specific node
+ * Finds specific node and zooms in
```

Comment:
similarly, add something a little more descriptive? like what parameters mean and what it returns